azure-storage-queue/src/main/scala/akka/stream/alpakka/azure/storagequeue/AzureQueueSourceStage.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.azure.storagequeue
+
+import com.microsoft.azure.storage.queue.{CloudQueue, CloudQueueMessage}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import akka.stream.{Attributes, Outlet, SourceShape}
+import scala.collection.mutable.Queue
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+final case class AzureQueueSourceSettings(
+    waitTimeSeconds: Int,
+    maxBufferSize: Int,
+    maxBatchSize: Int
+) {
+  require(maxBatchSize <= maxBufferSize, "maxBatchSize must be lower or equal than maxBufferSize")
+}
+
+object AzureQueueSourceSettings {
+  val default = AzureQueueSourceSettings(20, 100, 10)
+
+  /**
+   *  Java API
+   */
+  def create(waitTimeSeconds: Int, maxBufferSize: Int, maxBatchSize: Int): AzureQueueSourceSettings =
+    AzureQueueSourceSettings(waitTimeSeconds, maxBufferSize, maxBatchSize)
+}
+
+private[storagequeue] final class AzureQueueSourceStage(cloudQueue: CloudQueue, settings: AzureQueueSourceSettings)
```

Comment:
I switched to a factory. (Actually it should be thread-safe but are not guaranteed for future releases.)